### DISCUSSION TITLE
How to get revision information through plugin API ?

### DISCUSSION THREAD

**Discussion by lewis600** [ONLYOFFICE/DocumentServer#2488] [Q&A] [answered]
I didn't find a way to get revision information in the api documentation.

**Comment by askonev** [comment, marked as answer]
@lewis600 Greetings. May I ask for clarification on what is meant by `revision information`? ooxml documentation or screenshots of the editor would be a great help.

**Comment by lewis600** [comment]
See screenshot below：
<img width="858" alt="image" src="https://github.com/ONLYOFFICE/DocumentServer/assets/3233316/7068ba5e-2f52-4206-9bd1-13b342ca7152">

**Comment by askonev** [comment]
@lewis600 If I understood correctly it is necessary to get data from the set comment into the document. The [ApiComment](https://api.onlyoffice.com/docbuilder/textdocumentapi/apicomment) class is implemented for these tasks. Familiarize yourself with the usage examples and if you have any questions, ask them.

**Reply by lewis600** [reply]
I feel that we have different understandings of revisions. What I understand as revisions is that when I start revisions, a series of modifications to the document will be recorded, such as this example: [SetTrackRevisions(true)](https://api.onlyoffice.com/docbuilder/textdocumentapi/apidocument/rejectallrevisionchanges)
I can accept all revisions or reject all revisions like [AcceptAllRevisionChanges](https://api.onlyoffice.com/docbuilder/textdocumentapi/apidocument/acceptallrevisionchanges)
So I want to get all the revision information, just like getting all the comment information, provide a method GetAllRevisionChanges, such as [GetAllComments](https://api.onlyoffice.com/docbuilder/textdocumentapi/apidocument/getallcomments)

**Reply by askonev** [reply]
I created an ench in our private tracker `#65927`

**Reply by askonev** [reply]
@lewis600 A method that will work for your problem

https://api.onlyoffice.com/docbuilder/textdocumentapi/apidocument/getreviewreport

```js
var oDocument = Api.GetDocument();
var oParagraph1 = oDocument.GetElement(0);
oDocument.SetTrackRevisions(true);
oParagraph1.AddText("Reviewing documents");
oParagraph1.SetJc("center");
oParagraph1.SetFontSize(24);
oParagraph1.SetBold(true);
var oParagraph2 = Api.CreateParagraph();
oParagraph2.AddText("If you need to get review report, you can use Document Builder. The steps below will show how to do it.");
oDocument.Push(oParagraph2);
oDocument.SetTrackRevisions(false);
oDocument.AcceptAllRevisionChanges();
var oReviewReport = oDocument.GetReviewReport();
console.log(oReviewReport)
```

If you use the script without applying the changes, the object containing the changes will be displayed
![image](https://github.com/ONLYOFFICE/DocumentServer/assets/60688343/98fbeb36-8307-4aca-ad3b-736265e4f6ee)

**Reply by lewis600** [reply]
This looks good, it would be better if it could include paragraph information, I need to be able to quickly navigate to revisions.

**Reply by askonev** [reply]
Created a proposal to improve the method in our private repository `#66163`

**Comment by regent112** [comment]
Hello @lewis600. You can find the solution at [this](https://github.com/ONLYOFFICE/DocumentServer/issues/2546#issuecomment-4047373669)
